pg_types.go

```diff
@@ -0,0 +1,129 @@
+// Steve Phillips / elimisteve
+// 2017.05.18
+
+package main
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type PGClient struct {
+	BaseURL string
+}
+
+func NewPGClient(baseURL string) *PGClient {
+	return &PGClient{
+		BaseURL: baseURL,
+	}
+}
+
+func (cl *PGClient) Post(urlSuffix string, payload interface{}) (*http.Response, error) {
+	payloadb, err := json.Marshal(payload)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugf("POST'ing to %s: %s", urlSuffix, payloadb)
+
+	r := bytes.NewReader(payloadb)
+	req, _ := http.NewRequest("POST", cl.BaseURL+urlSuffix, r)
+	// req.Header.Add("Prefer", "return=representation")
+	req.Header.Add("Prefer", "return=none")
+	req.Header.Add("Content-Type", "application/json")
+
+	return http.DefaultClient.Do(req)
+}
+
+func (cl *PGClient) PostWanted(urlSuffix string, payload interface{}, statusWanted int) error {
+	resp, err := cl.Post(urlSuffix, payload)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != statusWanted {
+		body, _ := ioutil.ReadAll(resp.Body)
+
+		respjson := map[string]interface{}{}
+		err = json.Unmarshal(body, &respjson)
+		return fmt.Errorf(
+			"Got HTTP %v from PostgREST, wanted %v. Resp: %#v (err unmarshal: %v)",
+			resp.StatusCode, statusWanted, respjson, err)
+	}
+
+	return nil
+}
+
+func (cl *PGClient) Get(urlSuffix string) (*http.Response, error) {
+	log.Debugf("GET'ing from %s", urlSuffix)
```

Comment:
A few weeks ago, @ajvb set up `logrus`, so we now have log levels. That statement will only print to stdout if we're not in production, aka when we're dev'ing (yaaay)